### PR TITLE
doc: Add AWS MSK PrivateLink guide

### PR DESCRIPTION
**Pull request by bobbyiliev**
### Motivation

Adding a guide for the AWS MSK PrivateLink integration.

### CODE REVIEW

File: doc/user/content/ops/network-security/privatelink.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+### Steps
+
+1. #### Create Target Groups
+    Create [target groups](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/create-target-group.html) dedicated for each broker (in this example we will use: `b-1`, `b-2`) with the following details:
+
+    a. Target type as **IP address**.
+
+    b. Protocol as **TCP**.
+
+    c. Port as **9092,** or the port that you are using in case it is not 9092, e.g. 9094 for TLS or 9096 for SASL, etc.
```
Comment (joacoc) [inline doc/user/content/ops/network-security/privatelink.md:34]:
```suggestion
    c. For **Port**, modify the value to **9092,** or the port that you are using, e.g. 9094 for TLS or 9096 for SASL, etc.
```

File: doc/user/content/ops/network-security/privatelink.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+* A running [AWS MSK Cluster](/integrations/aws-msk)
+* A region enabled Materialize
+
+### Steps
+
+1. #### Create Target Groups
+    Create [target groups](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/create-target-group.html) dedicated for each broker (in this example we will use: `b-1`, `b-2`) with the following details:
+
+    a. Target type as **IP address**.
```
Comment (joacoc) [inline doc/user/content/ops/network-security/privatelink.md:30]:
```suggestion
    a. For **Target type**, choose ** IP addresses**.
```

File: doc/user/content/ops/network-security/privatelink.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+    b. Protocol as **TCP**.
```
Comment (joacoc) [inline doc/user/content/ops/network-security/privatelink.md:32]:
```suggestion
    b. For **Protocol**, choose **TCP**.
```

File: doc/user/content/ops/network-security/privatelink.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+    c. Port as **9092,** or the port that you are using in case it is not 9092, e.g. 9094 for TLS or 9096 for SASL, etc.
+
+    d. Make sure that the target group is in the same VPC as the MSK cluster.
```
Comment (joacoc) [inline doc/user/content/ops/network-security/privatelink.md:36]:
```suggestion
    d. For **VPC**, make sure you have the same as the MSK cluster.
```

File: doc/user/content/ops/network-security/privatelink.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+    e. Click next, and register the respective MSK broker to each target group by its IP address.
```
Comment (joacoc) [inline doc/user/content/ops/network-security/privatelink.md:38]:
```suggestion
    e. Click **Next**, and register the respective MSK broker to each target group by its IP address.
```

File: doc/user/content/ops/network-security/privatelink.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+    > Make sure to create a separate target group for each broker.
+
+1. #### Security Groups and Health Checks
+
+    Once the target groups have been created, make sure that the [health checks](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/target-group-health-checks.html) are passing and that the targets are reported as healthy.
```
Comment (joacoc) [inline doc/user/content/ops/network-security/privatelink.md:44]:
```suggestion
    After each target group creation, make sure that the [health checks](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/target-group-health-checks.html) are passing and that the targets are reported as healthy.
```

File: doc/user/content/ops/network-security/privatelink.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+    If you have set up a security group for your MSK cluster, you must ensure that the security groups allow traffic on both the listener port and the health check port.
+
+    **Remarks**:
+
+    a. Network Load Balancers do not have associated security groups. Therefore, the security groups for your targets must use IP addresses to allow traffic.
```
Comment (joacoc) [inline doc/user/content/ops/network-security/privatelink.md:50]: Should this remark be here or after point 3.? (Were you talking about NLBs)

**Review by joacoc — COMMENTED** [review]
Little tweaks about the first steps. I guess the styling on lists it is due to the comment in the Slack channel.

File: doc/user/content/ops/network-security/privatelink.md (hunk 7 above)
Comment (bobbyiliev) [reply doc/user/content/ops/network-security/privatelink.md:50]: Good point, but you need that information when you update your existing security groups (if any) right after setting up the target groups. The information itself is not very useful when creating the NLB itself.

**Review by joacoc — APPROVED** [review]

**Pull request merged**